agent/agentbootstrap/bootstrap.go

```diff
@@ -478,15 +478,18 @@ func (b *AgentBootstrap) initBootstrapMachine(
 		return nil, errors.Trace(err)
 	}
 
-	m, err := st.AddOneMachine(state.MachineTemplate{
-		Base:                    state.Base{OS: base.OS, Channel: base.Channel.String()},
-		Nonce:                   agent.BootstrapNonce,
-		Constraints:             stateParams.BootstrapMachineConstraints,
-		InstanceId:              stateParams.BootstrapMachineInstanceId,
-		HardwareCharacteristics: hardware,
-		Jobs:                    jobs,
-		DisplayName:             stateParams.BootstrapMachineDisplayName,
-	})
+	m, err := st.AddOneMachine(
+		&modelConfigShim{cfg: stateParams.ControllerModelConfig},
```

Comment:
This could be highly problematic as stateParams.ControllerModelConfig is not the final form of model config. It is just a suggestion of something that the client would like to see. I think this could be fine to use but not great.

@SimonRichardson are you able to confirm if there is a nice way for us to reach back into dqlite and pull the modelconfig after the bootstrap ops?

Reply:
This is only going to be reading two model config keys - `storage-default-block-source` and `storage-default-filesystem-source`.

Reply:
@tlm is indeed correct, we don't have access to dqlite. This is by design. We do not want to expose the database outside of the dependency engine, as it can lead to flakey tests and incorrect modeling. The solution is to move this to either the following:

https://github.com/juju/juju/blob/06a46d45d862f407611f58aed522a8ce3c7ea25b/agent/agentbootstrap/bootstrap.go#L269-L271

or

https://github.com/juju/juju/blob/06a46d45d862f407611f58aed522a8ce3c7ea25b/internal/worker/bootstrap/worker.go#L179-L265

Those are your official two options.

Reply:
I'll approve this change based on an item in the risk register and a comment stating this AddMachine call should be moved to the bootstrap worker.